### PR TITLE
Ensure all table names originate from Lhm::TableName

### PR DESCRIPTION
**Pull request by bbuchalter**
As you can see, the logic for describing table names is all over the place.
Trying to change the interfaces for these classes is beyond the scope of
the PR, as these interfaces are sadly coupled to the test suite.

However, we can at least have these classes delegate the logic for these
table names to something is well tested and centralized. This eliminates
the tests currently running on migration_spec.

Closes https://github.com/Shopify/lhm/issues/47

### CODE REVIEW

**Comment by bbuchalter** [comment]
This solves our primary problem. It doesn't change the length of the timestamp, because https://github.com/Shopify/lhm/issues/50, but at this point, we shouldn't care about that.

File: lib/lhm/table_name.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  class TableName
+    def initialize(original, time = Time.now)
+      @original = original
+      @time = time
+      @timestamp = Timestamp.new(time)
+    end
+
+    attr_reader :original
+
+    def archived
```
[the remark this answers is not in the document]
Comment (berlincount) [reply lib/lhm/table_name.rb]: but if that's the case, how does the test pass? I'm confused.

[the remark this answers is not in the document]
Comment (insom) [reply lib/lhm/table_name.rb]: @berlincount With three dots, ranges in Ruby are not inclusive:

```
irb(main):013:0> 'asdasdasdasdasdasdasdasdasdasdasdasdasdasdadasdasdasdasdasdasdasdasdasdadasd'[0...64].length
=> 64
```

With two dots they are. It's not my favourite syntax 😁

[the remark this answers is not in the document]
Comment (berlincount) [reply lib/lhm/table_name.rb]: what. the. beep.

Ruby magic strikes again.

**Pull request merged**